src/lib/Libnet/net_server.c

```diff
@@ -892,33 +853,32 @@ cleanup_conn(int cndx)
  *	We clear the cn_oncl field in the connection table to prevent any
  *	"special on close" functions from being called.
  *
- * @param[in] but - socket number
+ * @param[in] but - socket number to leave open
  *
  * @par	Note:
- *	free() the dynamically allocated data.
+ *	free() the dynamically allocated data
+ *
  */
-
 void
 net_close(int but)
 {
+	int sd;
 	conn_t *cp = NULL;
 
 	if (net_is_initialized == 0)
-		return;		/* not initialized, just return */
-
-	cp = (conn_t *)GET_NEXT(svr_allconns);
-	while(cp) {
-		int sock = cp->cn_sock;
-		cp = GET_NEXT(cp->cn_link);
-		if(sock != but) {
-			svr_conn[sock]->cn_oncl = NULL;
-			close_conn(sock);
+		return;
+
+	while((cp = (conn_t *)GET_NEXT(svr_allconns)) != NULL) {
```

Comment:
Was this broken prior to your change? This might be better structured as a for loop...

for (cp = (conn_t *)GET_NEXT(svr_allconns); cp != NULL; cp = (conn_t *)GET_NEXT(cp->cn_link)) {}

Reply:
No, not really broken. Was merely making it look nicer. I agree that changing to a for loop would be the best.

Reply:
Oops. On second thoughts, I found my change wrong, and also the suggested for loop will be a problem. This is because close_conn() which is called inside the loop body deletes a node in the link list. This leaves cp a dangling pointer, which in the old code they handled properly. So restoring the old code.

Reply:
Works for me.